packages/data-stores/src/automated-transfer-eligibility/utilities.ts

```diff
@@ -0,0 +1,52 @@
+import { statusMapping } from './constants';
+import { TransferEligibilityWarning, TransferEligibilityHold, TransferEligibility } from './types';
+
+export function getEligibilityHolds(
+	transferEligibility: TransferEligibility | null
+): TransferEligibilityHold[] | null {
+	if ( ! transferEligibility?.errors ) {
+		return null;
+	}
+
+	return transferEligibility.errors.reduce< TransferEligibilityHold[] >( ( errs, { code } ) => {
+		if ( code in statusMapping ) {
+			// At this point, we know code is a key of statusMapping, so we help TS understand that.
+			errs.push( statusMapping[ code as keyof typeof statusMapping ] );
+		}
+		return errs;
```

Comment:
Nice simplification to hold only the statuses since we didn't really need the messages.